### ISSUE TITLE
[Storage] Add option to store a hash beside a file

### ISSUE THREAD

**Issue by aurelien-reeves-scalingo** [Scalingo/go-utils#395]
At the moment, when storing ciphered data on external storage, we keep a hash in database to be able to check streamed data against it after it has been downloaded.

Maybe there may be some ways to add a check of the hash before starting to stream the data - **in addition** to the other checks, **not in replacement**.

Maybe the APIs of our storage provider may help retrieving the hash using the same algorithm than ours of a given file?

In last resort we could store the hash beside the file

**Comment by aurelien-reeves-scalingo** [comment]
IIRC the idea is to check the hash of a file hosted on an external storage before downloading it, in addition to check the hash once it has been downloaded.

We keep the hash in our database. We could request the hash to the external hosting provider to compare it to the one we have stored in the database. If it is the hash is wrong, we would not download the file at all.

**Comment by SCedricThomas** [comment]
Won't do as not needed for now
